lib/sqlalchemy/dialects/mssql/base.py

```diff
@@ -2911,6 +2911,44 @@ def do_release_savepoint(self, connection, name):
         ]
     )
 
+    def get_sql_server_variant(self, dbapi_connection) -> str:
+        """
+        Queries SQL Server system views to determine if the given connection
+        is to a normal SQL Server instance or a PDW-based instance.
+
+        Parameters
+        ------------
+        dbapi_connection -- the underlying DBAPI Connection
+
+        Returns
+        --------
+        A string indicating the server variant, "sql_server" or "pdw"
+        """
+        cursor = dbapi_connection.cursor()
+        edition = "sql_server"
+        error = None
+
+        try:
```

Comment:
Sounds like a good idea to me.

Reply:
oh, sure.  this query is usualyl done only once anyway on first connect unless someone is using the Connection API, which is probably never used, but yes it could be memoized.   mostly I want to get somethign committed so we are ready for 1.4.40 on this